modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -656,9 +649,10 @@ private CompletableFuture<SchemaPlus> waitForActualSchema(String schemaName, Hyb
         }
     }
 
-    private AsyncSqlCursor<InternalSqlRow> executePlan(
+    private CompletableFuture<AsyncSqlCursor<InternalSqlRow>> executePlan(
             QueryTransactionWrapper txWrapper,
             BaseQueryContext ctx,
+            PrefetchCallback callback,
```

Comment:
actually, we can't. I passed it explicitly in order to preserve type `PrefetchCallback` since I need a future, not the callback to notify. Another options are to do a cast from `QueryPrefetchCallback` to `PrefetchCallback`, and to extend QueryPrefetchCallback exposing a future.

I prefer to pass callback explicitly, this approach seams to be lesser evil